src/dotnet-uninstall/Program.cs

```diff
@@ -2,14 +2,49 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Runtime.InteropServices;
 
 namespace Microsoft.DotNet.Tools.Uninstall
 {
     internal class Program
     {
-        static void Main(string[] args)
+        internal static readonly bool RunningOnWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+        internal static readonly bool RunningOnOSX = RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
+        internal static readonly bool RunningOnLinux = RuntimeInformation.IsOSPlatform(OSPlatform.Linux);
+
+        static int Main(string[] args)
         {
-            Console.WriteLine(Messages.message);
+            var rootCommand = new RootCommand("dotnet-uninstall");
+
+            var listCommand = new Command("list");
+            var uninstallCommand = new Command("uninstall");
+            rootCommand.Add(listCommand);
+            rootCommand.Add(uninstallCommand);
+
+            listCommand.Handler = CommandHandler.Create(() =>
+            {
+                ExecuteListCommand();
+            });
+
+            uninstallCommand.Handler = CommandHandler.Create(() =>
+            {
+                Console.WriteLine("invoked uninstall subcommand"); // TODO: implement uninstall
```

Comment:
minimize todo. this can track in "issues". And we can just point "not support" for now

Reply:
done